plugin/storage/cassandra/spanstore/operation_names_test.go

```diff
@@ -118,28 +139,40 @@ func TestOperationNamesStorageGetServices(t *testing.T) {
 		return true
 	})
 	matchEverything := mock.MatchedBy(func(v []interface{}) bool { return true })
-	for _, expErr := range []error{nil, scanError} {
-		withOperationNamesStorage(writeCacheTTL, func(s *operationNameStorageTest) {
+	for _, test := range []struct {
+		name          string
+		schemaVersion string
+		expErr        error
+	}{
+		{name: "test old schema without error", schemaVersion: previousVersion, expErr: nil},
+		{name: "test old schema with scan error", schemaVersion: previousVersion, expErr: scanError},
+		{name: "test new schema without error", schemaVersion: latestVersion, expErr: nil},
+		{name: "test new schema with scan error", schemaVersion: latestVersion, expErr: scanError},
+	} {
+		fmt.Printf(test.name)
+		withOperationNamesStorage(0, test.schemaVersion, func(s *operationNameStorageTest) {
 			iter := &mocks.Iterator{}
 			iter.On("Scan", matchOnce).Return(true)
 			iter.On("Scan", matchEverything).Return(false) // false to stop the loop
-			iter.On("Close").Return(expErr)
+			iter.On("Close").Return(test.expErr)
 
 			query := &mocks.Query{}
 			query.On("Iter").Return(iter)
 
-			s.session.On("Query", mock.AnythingOfType("string"), []interface{}{"service-a"}).Return(query)
-
+			s.session.On("Query", mock.AnythingOfType("string"), mock.Anything).Return(query)
 			services, err := s.storage.GetOperations("service-a")
-			if expErr == nil {
+			if test.expErr == nil {
 				assert.NoError(t, err)
-				// expect empty string because mock iter.Scan(&placeholder) does not write to `placeholder`
-				assert.Equal(t, []string{""}, services)
+				if test.schemaVersion == previousVersion {
+					// expect one empty operation result because mock iter.Scan(&placeholder) does not write to `placeholder`
+					assert.Equal(t, []string{""}, services)
+				} else {
+					assert.Equal(t, []string{}, services)
```

Comment:
it is not clear to me why in the first case there's an empty string in the array, and in the second case an empty array

Reply:
I actually fixed that in the second PR, it's due to the initialization of the mock. Going to move it here.